AMW_angular/io/src/app/deployment/environment.ts

```diff
@@ -6,3 +6,28 @@ export interface Environment {
   selected: boolean;
   disabled: boolean;
 }
+
+export class EnvironmentTree {
```

Comment:
Personally - I don't like classes.

I'd probably just define a type and thats it. If it matches the Response from the backend, you won't need a constructor. If necessary, you could create a simple function that converts between the API and your type.